shared/src/types/key/ed25519.rs

```diff
@@ -460,9 +572,82 @@ impl From<SecretKey> for ed25519_dalek::SecretKey {
     }
 }
 
-impl Display for PublicKeyHash {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0)
+impl From<ed25519_dalek::Signature> for Signature {
+    fn from(sig: ed25519_dalek::Signature) -> Self {
+        Self(sig)
+    }
+}
+
+impl From<Signature> for ed25519_dalek::Signature {
+    fn from(sig: Signature) -> Self {
+        sig.0
+    }
+}
+
+impl From<ed25519_dalek::Keypair> for Keypair {
+    fn from(keypair: ed25519_dalek::Keypair) -> Self {
+        Self {
+            secret: keypair.secret.into(),
+            public: keypair.public.into(),
+        }
+    }
+}
+
+impl From<Keypair> for ed25519_dalek::Keypair {
+    fn from(keypair: Keypair) -> Self {
+        Self {
+            secret: keypair.secret.into(),
+            public: keypair.public.into(),
+        }
+    }
+}
+
+impl Signer<Signature> for Keypair {
```

Comment:
all the impls from here below are just wrappers for the library code on newtypes